hazelcast-utils/pom.xml

```diff
@@ -0,0 +1,98 @@
+    <modelVersion>4.0.0</modelVersion>
+
+    <!--
+    The purpose of this module is to collect support utility functionality that is common
+    to both client and server. The utils module should remain relatively slim; things like
+    networking should be placed in their own modules. With proper modularization we can
+    start to reduce the coupling in the code.
+
+    Functionality that is specific to clients or members should not be moved into this module.
+    -->
+
+    <name>hazelcast-utils</name>
+    <artifactId>hazelcast-utils</artifactId>
+    <description>The Hazelcast commons module.</description>
+    <packaging>jar</packaging>
+
+    <parent>
+        <groupId>com.hazelcast</groupId>
+        <artifactId>hazelcast-root</artifactId>
+        <version>5.4.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <properties>
+        <!-- Needed for CheckStyle -->
+        <main.basedir>${project.parent.basedir}</main.basedir>
+     </properties>
+
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+            </resource>
+        </resources>
+
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <configuration>
+                    <archive>
+                        <index>true</index>
+                        <compress>true</compress>
+                        <manifest>
+                            <addClasspath>false</addClasspath>
+                            <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
+                            <addDefaultSpecificationEntries>true</addDefaultSpecificationEntries>
+                        </manifest>
+                    </archive>
+                </configuration>
+                <executions>
+                    <execution>
+                        <id>test-jar</id>
+                        <goals>
+                            <goal>test-jar</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <!-- We must generate a -javadoc JAR file to publish on Maven Central -->
+                        <id>empty-javadoc-jar</id>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                        <configuration>
+                            <classifier>javadoc</classifier>
+                            <classesDirectory>${project.basedir}/src/main/javadoc</classesDirectory>
+                            <excludes>
+                                <exclude>.*</exclude>
+                            </excludes>
+                        </configuration>
+                    </execution>
+                </executions>
```

Comment:
Do we need this? `com.hazelcast.logger.*` is public API and the javadoc jar should be created. We use this only if we have a module that only includes e.g. `.*.impl.*` classes.

Reply:
Not sure if we need it. I copied it from a different module. Let me have another look at the pom.xml.